libs/auth-api-lib/src/lib/resources/delegation-resources.service.ts

```diff
@@ -44,16 +44,37 @@ export class DelegationResourcesService {
 
   async findAllDomains(
     user: User,
-    language?: string,
-    direction?: DelegationDirection,
+    {
+      language,
+      direction,
+      domainNames,
+      supportsDelegations,
+    }: {
+      language?: string
+      direction?: DelegationDirection
+      domainNames?: string[]
+      supportsDelegations?: boolean
+    },
   ): Promise<Domain[]> {
+    const onlyDelegations =
+      supportsDelegations || direction === DelegationDirection.OUTGOING
+
+    const domainNameFilter: WhereOptions<Domain> = domainNames
+      ? { name: domainNames }
+      : {}
+
     const domains = await this.domainModel.findAll({
-      where: and(...this.apiScopeFilter({ user, prefix: 'scopes', direction })),
+      where: onlyDelegations
+        ? and(
+            ...this.apiScopeFilter({ user, prefix: 'scopes', direction }),
+            domainNameFilter,
+          )
+        : domainNameFilter,
       include: [
```

Comment:
Should skip this whole join tree unless `onlyDelegations`?

Reply:
It is already doing that, when `onlyDelegations=false` then it only adds the `name: domainNames` if provided.
```
onlyDelegations
        ? and(
            ...this.apiScopeFilter({ user, prefix: 'scopes', direction }),
            domainNameFilter,
          )
        : domainNameFilter,
```

Reply:
You meant the `include`, yes considered skipped.